### PR TITLE
replace zuul proxy with spring cloud gateway 2.x. in webapp module.

### PR DESCRIPTION
**Pull request by JaredTan95**
provide a new solution for rocketbot ui backend(oap) proxy.

- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes https://github.com/apache/skywalking/issues/7122.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/CHANGES.md).

### CODE REVIEW

**Comment by JaredTan95** [comment]
image

works fine in my local test

File: apm-webapp/src/main/resources/application.yml [hunk 1]
```diff
@@ -17,28 +17,28 @@
...
-    listOfServers: 127.0.0.1:12800
-
 spring:
-  resources:
-    add-mappings: false
+  cloud:
+    gateway:
+      routes:
+        - id: oap-route
+          uri: lb://oap-service
```
Comment (wu-sheng) [inline apm-webapp/src/main/resources/application.yml:25]: Where does this mean?

File: apm-webapp/pom.xml [hunk 2]
```diff
@@ -29,18 +30,18 @@
...
         <apache-httpclient.version>4.5.3</apache-httpclient.version>
-        <spring-cloud-dependencies.version>Edgware.SR1</spring-cloud-dependencies.version>
+        <spring-cloud-dependencies.version>2020.0.3</spring-cloud-dependencies.version>
         <frontend-maven-plugin.version>1.11.0</frontend-maven-plugin.version>
         <logback-classic.version>1.2.3</logback-classic.version>
         <jackson-version>2.12.2</jackson-version>
         <yaml.version>1.28</yaml.version>
-        <netty.version>4.1.65.Final</netty.version>
-        <tomcat.version>8.5.66</tomcat.version>
+        <tomcat.version>9.0.48</tomcat.version>
```
Comment (wu-sheng) [inline apm-webapp/pom.xml:44]: dist/LICENSE should be updated accordingly.

**Review by hanahmily — CHANGES_REQUESTED** [review]
Could you update https://github.com/apache/skywalking/blob/master/docs/en/setup/backend/ui-setup.md correspondingly?

File: apm-webapp/src/main/resources/application.yml (hunk 1 above)
Comment (JaredTan95) [reply apm-webapp/src/main/resources/application.yml:25]: `lb:` represents client-side  `load balance`

**Comment by JaredTan95** [comment]
> Could you update https://github.com/apache/skywalking/blob/master/docs/en/setup/backend/ui-setup.md correspondingly?

and helm deployments also need to be updated after merged, right.

File: apm-webapp/pom.xml (hunk 2 above)
Comment (JaredTan95) [reply apm-webapp/pom.xml]: I did not found `springboot/cloud` license under `dist/LICENSE`, I'm not sure if I need to add it

Comment (kezhenxu94) [reply apm-webapp/pom.xml]: https://github.com/apache/skywalking/blob/669fe1593c8cd3591652c13d1088c72d2c45ab89/dist-material/release-docs/LICENSE#L254

File: dist-material/release-docs/LICENSE [hunk 3]
```diff
@@ -270,9 +272,7 @@ The text of each license is the standard Apache 2.0 license.
     Apache: commons-io 2.4: https://github.com/apache/commons-io, Apache 2.0
     Apache: commons-compress 1.20: https://github.com/apache/commons-compress, Apache 2.0
     Apache: commons-collections4 4.4: https://mvnrepository.com/artifact/org.apache.commons/commons-collections4, Apache 2.0
-    Apache: tomcat 8.5.66: https://github.com/apache/tomcat/tree/trunk, Apache 2.0
     Apache: freemarker 2.3.28: https://github.com/apache/freemarker, Apache 2.0
-    netty 4.1.65: https://github.com/netty/netty/blob/4.1/LICENSE.txt, Apache 2.0
```
Comment (kezhenxu94) [inline dist-material/release-docs/LICENSE:275]: This is still used in oap-server

**Comment by wu-sheng** [comment]
Please make CI passed :)

**Comment by JaredTan95** [comment]
> Please make CI passed :)

yes, i am still workong on it.

**Comment by JaredTan95** [comment]
@wu-sheng @kezhenxu94 I noticed that Istio+ALS e2e needs skywalking-kubernetes to be updated within https://github.com/apache/skywalking-kubernetes/pull/74, but it need this PR merged first. :-P

**Comment by kezhenxu94** [comment]
> @wu-sheng @kezhenxu94 I noticed that Istio+ALS e2e needs skywalking-kubernetes to be updated within [apache/skywalking-kubernetes#74](https://github.com/apache/skywalking-kubernetes/pull/74), but it need this PR merged first. :-P

Can we merge https://github.com/apache/skywalking-kubernetes/pull/74 first?

**Comment by JaredTan95** [comment]
> > @wu-sheng @kezhenxu94 I noticed that Istio+ALS e2e needs skywalking-kubernetes to be updated within [apache/skywalking-kubernetes#74](https://github.com/apache/skywalking-kubernetes/pull/74), but it need this PR merged first. :-P
> 
> Can we merge [apache/skywalking-kubernetes#74](https://github.com/apache/skywalking-kubernetes/pull/74) first?

If skywalking-kubernetes does not dependency this PR's docker image, I think we could merge [apache/skywalking-kubernetes#74](https://github.com/apache/skywalking-kubernetes/pull/74) first.

**Comment by kezhenxu94** [comment]
> > > @wu-sheng @kezhenxu94 I noticed that Istio+ALS e2e needs skywalking-kubernetes to be updated within [apache/skywalking-kubernetes#74](https://github.com/apache/skywalking-kubernetes/pull/74), but it need this PR merged first. :-P
> > 
> > 
> > Can we merge [apache/skywalking-kubernetes#74](https://github.com/apache/skywalking-kubernetes/pull/74) first?
> 
> If skywalking-kubernetes does not dependency this PR's docker image, I think we could merge [apache/skywalking-kubernetes#74](https://github.com/apache/skywalking-kubernetes/pull/74) first.

It doesn't, you can merge that first, and update https://github.com/apache/skywalking/blob/04bb6673b8e7d0df07ba6fdd09ba78ff3ac89148/.github/workflows/e2e.istio.yaml#L98 in this PR, after this PR gets merged, then update skywalking-kubernets 
https://github.com/apache/skywalking-kubernetes/blob/49f9e3cef4e995dde9b77fb3b24876dd2d9c6c12/.github/workflows/e2e.ci.yaml#L46 to verify

**Comment by JaredTan95** [comment]
ok, thanks

File: .github/workflows/e2e.istio.yaml [hunk 4]
```diff
@@ -95,7 +95,7 @@ jobs:
         run: |
           git clone https://github.com/apache/skywalking-kubernetes.git
           cd skywalking-kubernetes
-          git reset --hard 2fdcdc3bb39496bf49626755e8f60c998be5f587
+          git reset --hard 6d5897616ce30ebb1706c0cf566ac36f733d93e0
```
Comment (kezhenxu94) [inline .github/workflows/e2e.istio.yaml:98]: Line 240 should be updated too

**Review by kezhenxu94 — DISMISSED** [review]
LGTM, thanks!!

**Comment by wu-sheng** [comment]
@hanahmily We need your recheck.

**Comment by JaredTan95** [comment]
all e2e passed, finnally. :-)

**Review by wu-sheng — APPROVED** [review]

**Review by hanahmily — APPROVED** [review]
LGTM

**Pull request merged**